### ISSUE TITLE
TypeScript issues with Node 16

### ISSUE THREAD

**Issue by treyturner** [dlenroc/node-testrail-api#73]
Hi again,
I'm having trouble using your library with Node 16 and TypeScript. I'm working in a large monorepo so upgrading to Node 18 is not currently an option.

Including `@types/node` `16.11.20` in my dependencies and trying to type-check a script that uses your library is throwing these errors:

```
$ tsc --noEmit
node_modules/@dlenroc/testrail/src/internal/request.ts:70:34 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

70     return Promise.reject(signal.reason);
                                    ~~~~~~

node_modules/@dlenroc/testrail/src/internal/request.ts:86:22 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

86       reject(signal?.reason);
                        ~~~~~~

node_modules/@dlenroc/testrail/src/TestRail.ts:504:39 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

504       return Promise.reject(ctxSignal.reason || options.signal.reason);
                                          ~~~~~~

node_modules/@dlenroc/testrail/src/TestRail.ts:504:64 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

504       return Promise.reject(ctxSignal.reason || options.signal.reason);
                                                                   ~~~~~~

node_modules/@dlenroc/testrail/src/TestRail.ts:510:70 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

510     const onAbort = () => { clearAbort(); controller.abort(ctxSignal.reason || optionsSignal.reason); };
                                                                         ~~~~~~

node_modules/@dlenroc/testrail/src/TestRail.ts:510:94 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

510     const onAbort = () => { clearAbort(); controller.abort(ctxSignal.reason || optionsSignal.reason); };
                                                                                                 ~~~~~~


Found 6 errors in 2 files.

Errors  Files
     2  node_modules/@dlenroc/testrail/src/internal/request.ts:70
     4  node_modules/@dlenroc/testrail/src/TestRail.ts:504
error Command failed with exit code 2.
```

This seems to be because the `AbortSignal` interface changed between Node 16 and 18. I managed to pass your library the signal by hand (as opposed to using `AbortSignal.timeout()` as discussed in #70 since this isn't available in Node 16), so all my code passes type checks, but TypeScript seems to be checking the `.ts` files in your library despite my ignoring `node_modules` in `tsconfig.json`.

I created a minimum reproducible example here: https://github.com/treyturner/testrail-debug. You can see the problem I'm having by running `yarn type-check`. Is this a misconfiguration on my part?

I'm pretty inexperienced when it comes to JS/TS build configuration and package bundling; I see you ship both a `dist` folder and a `types.d.ts` file in your package. Are the `.ts` files needed? It occurs to me that if they weren't included, TypeScript wouldn't take issue with the package.

Thanks again for your time and work on the library.

**Comment by dlenroc** [comment]
that seems to be a typing issue (node js 16 has `reason` property but is not documented in typing)

you can get around this by creating a `src/types.d.ts` file with the following content

```typescript
interface AbortSignal {
  readonly reason: any;
}
```

I will either embed typing in the library, use `any`, or ship `d.ts` (which supports `skipLibCheck: true`) to make type check pass on multiple node versions (but will do that later today)

**Comment by dlenroc** [comment]
A fix for this issue is included in version 1.9.1 (thanks for the issue and reproducible example it was a great help).
